napari/utils/_proxies.py

```diff
@@ -100,8 +100,12 @@ def __getattr__(self, name: str):
             typ = type(self.__wrapped__).__name__
 
             self._private_attr_warning(name, typ)
+        with warnings.catch_warnings(record=True) as cx_manager:
```

Comment:
Aren't we trying to avoid warning because of a bug upstream that reset warnings ?

Otherwise +1

Reply:
#6396 is workaround for this problem. 
Bug will be present if someone will still use PublicOnlyProxy without custom warning handle.